### PR TITLE
test: migrate route_export_test e2e

### PR DESCRIPTION
**Pull request by zaunist**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Migrate old route_export_test e2e to ginkgo implementation.

Please update this section with detailed description.

**Related issues**

Part of #2197
fix/resolve #0001

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by bzp2010** [comment]
This PR is so long that I think it is necessary to split it up. In the meantime, I remember that we already have tests related to routing import and export in `e2enew`, so you can check if there are overlapping parts.

**Comment by zaunist** [comment]
> This PR is so long that I think it is necessary to split it up. In the meantime, I remember that we already have tests related to routing import and export in `e2enew`, so you can check if there are overlapping parts.

OK, I will check again.

**Comment by zaunist** [comment]
Hi, @bzp2010 ，When I checked the repository, I found that this [file](https://github.com/apache/apisix-dashboard/blob/master/api/test/e2enew/migrate/migrate_test.go) has tests for import and export, but only some of them, so I think the tests are necessary for this commit.

**Comment by zaunist** [comment]
Of course, some of these test cases do not match the current version, so if this commit is necessary, I'll fix the ones that are wrong as soon as possible.

**Comment by zaunist** [comment]
cc @bzp2010 @nic-chen @starsz @juzhiyuan @bisakhmondal

File: api/test/e2enew/route/route_export_test.go [hunk 1]
```diff
@@ -0,0 +1,2490 @@
...
+var _ = ginkgo.Describe("Route", func() {
+	ginkgo.Context("test route export data empty", func() {
+		ginkgo.It("Export route when data is empty", func() {
+			base.RunTestCase(base.HttpTestCase{
+				Object:       base.ManagerApiExpect(),
+				Method:       http.MethodGet,
+				Path:         "/apisix/admin/export/routes",
+				Headers:      map[string]string{"Authorization": base.GetToken()},
+				ExpectStatus: http.StatusOK,
+				ExpectBody:   "{\"code\":10000,\"message\":\"Route data is empty, cannot be exported\",\"data\":null",
```
Comment (starsz) [inline api/test/e2enew/route/route_export_test.go:39]: The logic is strange. Why do we not allow export of the empty route? (This is another question)

File: api/test/e2enew/route/route_export_test.go [hunk 2]
```diff
@@ -0,0 +1,2490 @@
...
+				 "build": "16",
+				 "env": "production",
+				 "version": "v2"
+			 },
+			 "enable_websocket": true
+			 `
+		serviceStrS1 = replaceStr(serviceStrS1)
+
+		exportStrR3 := `{
+							 "components": {},
```
Comment (starsz) [inline api/test/e2enew/route/route_export_test.go:389]: Can you improve the style?

Comment (zaunist) [reply api/test/e2enew/route/route_export_test.go:389]: OK

File: api/test/e2enew/route/route_export_test.go (hunk 1 above)
Comment (zaunist) [reply api/test/e2enew/route/route_export_test.go:39]: Code 10000, this is an error message, is it right? So it's a correct act?

**Review by starsz — COMMENTED** [review]
Why the PR has too many file changes?

**Comment by juzhiyuan** [comment]
image

Maybe due to the force-push?

**Review by starsz — COMMENTED** [review]
Why the PR has too many changes?

**Comment by zaunist** [comment]
> Why the PR has too many changes?

Sorry, I will fix it

**Review by starsz — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**